From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: psacal <2658842706@qq.com>
Date: Thu, 3 Jun 2026 00:00:00 +0800
Subject: [PATCH] drm/panel: himax-hx83121a: expose panel orientation

Call drm_connector_set_panel_orientation() so that the "panel orientation"
DRM connector property is created. This allows mutter (GNOME 50+) to
enable PanelOrientationManaged for automatic screen rotation.

Signed-off-by: psacal <2658842706@qq.com>
---
 drivers/gpu/drm/panel/panel-himax-hx83121a.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/panel/panel-himax-hx83121a.c b/drivers/gpu/drm/panel/panel-himax-hx83121a.c
index 028b615c6..0000000000 100644
--- a/drivers/gpu/drm/panel/panel-himax-hx83121a.c
+++ b/drivers/gpu/drm/panel/panel-himax-hx83121a.c
@@ -20,6 +20,7 @@
 #include <drm/drm_mipi_dsi.h>
 #include <drm/drm_modes.h>
 #include <drm/drm_panel.h>
+#include <drm/drm_connector.h>

 #include <video/mipi_display.h>

@@ -182,6 +183,8 @@ static int himax_get_modes(struct drm_panel *panel,
 	connector->display_info.width_mm = desc->width_mm;
 	connector->display_info.height_mm = desc->height_mm;
 	connector->display_info.bpc = desc->bpc;
+	drm_connector_set_panel_orientation(connector,
+					     DRM_MODE_PANEL_ORIENTATION_NORMAL);

 	return num_modes;
 }
--
2.43.0
